docs/reference/sql/endpoints/rest.asciidoc

```diff
@@ -106,6 +106,21 @@ s|Description
 
 |===
 
+The `CSV` format accepts the following additional formatting URL query attributes:
+
+* `header`
++
+Controls if the CSV header should be included in the response or not. It can take one of the following values:
++
+   - `present` (default): include the header;
+   - `absent`: exclude the header.
```

Comment:
> Why not using a true/false kind of approach with the csv header?

The `absent`/`present` values [were coded in before](https://github.com/elastic/elasticsearch/blob/74c9e5673561eed60700edc5043c5618dad26b2b/x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/TextFormat.java#L186). And my guess is that the values have been chosen to be in-line with the [CT header values](https://github.com/elastic/elasticsearch/blob/74c9e5673561eed60700edc5043c5618dad26b2b/x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/TextFormat.java#L136).

> it was a way of distinguishing between the first page of the response and the subsequent pages of it

I see. ..actually how would a subsequent page be requested with a URL attribute in CSV format? I think we have no `OFFSET`, neither at API, nor at SQL level, do we?
I also found no tests covering this scenario or the use of the `header` attribute.

And these two points above...

> why did you expose this request parameter in the first place?

...let me to believe that it might be an intended feature, but no[t / longer] correctly functioning.

And I think it would be good to have it, CSV handling applications generally offer such functionality. So irrespective of the original intent, is there any concern exposing it?

> Also, the initial request this PR is fixing refers only to the delimiter itself.

True. Would it make sense to split the PR (vs. maybe extending this PR's description)?

Reply:
> ...let me to believe that it might be an intended feature, but no[t / longer] correctly functioning.
My 2c on this: if it's requested, we can discuss about adding it. If there is no demand, we don't complicate things.

> I see. ..actually how would a subsequent page be requested with a URL attribute in CSV format? I think we have no OFFSET, neither at API, nor at SQL level, do we?
The first response that comes from ES contains a header called `cursor` to be used for a subsequent call to get the next page of rows. The URL remains the same, but the second call doesn't have a `query` in it, but a `cursor`.

Reply:
> My 2c on this: if it's requested, we can discuss about adding it. If there is no demand, we don't complicate things.

Sure, I can take it out. But just to understand how to take it out:

> it was a way of distinguishing between the first page of the response and the subsequent pages of it

How is this distinguishing being done, since I believe there's no way to specify a `header` URL attribute currently, since it's not "consumed" by the `RestSqlQueryAction` and so using it would generate a `"request [/_sql] contains unrecognized parameter: [header]"` error. Should I remove it entirely then? Sorry if missing something.

> The first response that comes from ES contains a header called cursor to be used for a subsequent call to get the next page of rows.

Right, it's inserted by [`RestSqlQueryAction`](https://github.com/elastic/elasticsearch/blob/380ff0b0d44c0e0036a8934d2b146ee08645ba26/x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/RestSqlQueryAction.java#L117), should have spotted it before. Thanks.
Is this documented anywhere though? I couldn't find it on the [REST formats page](https://www.elastic.co/guide/en/elasticsearch/reference/master/sql-rest-format.html). I can add it if not.

Reply:
> Sure, I can take it out. But just to understand how to take it out:

Just do not expose it as a request parameter. Leave the rest of the logic unchanged and this PR should only address the delimiter part.

> How is this distinguishing being done, since I believe there's no way to specify a header URL attribute currently, since it's not "consumed" by the RestSqlQueryAction and so using it would generate a "request [/_sql] contains unrecognized parameter: [header]" error. Should I remove it entirely then? Sorry if missing something.

I don't think it's actually being used, indeed. I see logic that's based on the presence of the cursor or not, not much about the `header` being present or not in the request...

> Right, it's inserted by RestSqlQueryAction, should have spotted it before. Thanks.
Is this documented anywhere though? I couldn't find it on the REST formats page. I can add it if not.

It's not documented for other formats (except json and txt), either. I think a simple statement change in [this page](https://www.elastic.co/guide/en/elasticsearch/reference/current/sql-pagination.html) also mentioning csv and tsv should be enough (ie, no examples).

Reply:
I've reverted the changes here and opened #60271, since this is clearly a (very minor) bug.